### PR TITLE
Update glaze add mass

### PR DESCRIPTION
**Pull request by jbbjarnason**
- Add mass ipc thingy
- Update glaze 
- Move vcpkg dependencies to binary dir
- Comment out extra lld setter
- Remove boost lexical cast from tfcctl
- Refactor tfcctl heavily
- Read only is read only in json schema
- std array is fixed size in json schema

### CODE REVIEW

File: libs/stx/inc/public/tfc/stx/glaze_meta.hpp [hunk 1]
```diff
@@ -102,18 +106,35 @@ static constexpr auto make_name() -> std::string_view {
...
+  static constexpr std::string_view delimiter{ ", " };
+  static constexpr std::string_view name{
+    tfc::stx::string_view_join_v<prefix, name_v<value_t>, delimiter, name_v<error_t>, postfix>
+  };
+  static constexpr auto value{ [](auto&& self) -> auto& {
+    // todo this does not support error case
+    return self.value();
+  } };
+};
+// static_assert(glz::detail::nullable_t<std::expected<int, std::error_code>>);
```
Comment (jbbjarnason) [inline libs/stx/inc/public/tfc/stx/glaze_meta.hpp:129]:
```suggestion
```

File: libs/ipc/src/dbus_client_iface.cpp [hunk 2]
```diff
@@ -63,11 +64,14 @@ auto ipc_manager_client::signals(std::function<void(std::vector<signal> const&)>
       consts::signals_property.data(),
       [captured_handler = std::move(handler)](const boost::system::error_code& error, const std::string& response) {
         if (error) {
+          fmt::println(stderr, "Get property signals handler error: {}", error.message());
```
Comment (omarhogni) [inline libs/ipc/src/dbus_client_iface.cpp:67]: This should propably use a logger.

File: libs/ipc/src/dbus_client_iface.cpp [hunk 3]
```diff
@@ -63,11 +64,14 @@ auto ipc_manager_client::signals(std::function<void(std::vector<signal> const&)>
...
           return;
         }
         auto signals = glz::read_json<std::vector<signal>>(response);
         if (signals) {
           captured_handler(signals.value());
+        } else {
+          fmt::println(stderr, "Get property signals parse error: {}", glz::format_error(signals.error(), response));
```
Comment (omarhogni) [inline libs/ipc/src/dbus_client_iface.cpp:74]: this to

File: libs/ipc/src/dbus_client_iface.cpp [hunk 4]
```diff
@@ -1,5 +1,6 @@
 #include <tfc/ipc/details/dbus_client_iface.hpp>
 
+#include <fmt/core.h>
```
Comment (omarhogni) [inline libs/ipc/src/dbus_client_iface.cpp:3]: remove if switch to logger.

**Review by omarhogni — CHANGES_REQUESTED** [review]

File: cmake/tfc_config.cmake [hunk 5]
```diff
@@ -6,8 +6,8 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
   # enable full llvm assembler, byte code gen, linking optimisations
-  add_compile_options( -fuse-ld=lld )
-  add_link_options( -fuse-ld=lld )
+#  add_compile_options( -fuse-ld=lld )
+#  add_link_options( -fuse-ld=lld )
```
[the remark this answers is not in the document]
Comment (jbbjarnason) [reply cmake/tfc_config.cmake]: has been removed

File: libs/ipc/inc/public/tfc/ipc/glaze_meta.hpp [hunk 6]
```diff
@@ -37,9 +37,31 @@ struct meta<tfc::ipc::details::type_e> {
...
+template <>
+struct meta<tfc::ipc::details::mass_error_e> {
+  using enum tfc::ipc::details::mass_error_e;
+  // clang-format off
+  static auto constexpr value{ glz::enumerate(
+    "no_error", no_error, "Not an error",
+    "cell_fault", cell_fault, "Load cell fault",
+    "module_fault", module_fault, "General weigher module fault",
+    "power_failure", power_failure, "Supply voltage too low",
+    "over_range", over_range, "Over max value, please take care of the load cell",
```
[the remark this answers is not in the document]
Comment (jbbjarnason) [reply libs/ipc/inc/public/tfc/ipc/glaze_meta.hpp]: removed the extra info, now only Over max value

File: libs/dbus_util/inc/public/tfc/dbus/sdbusplus_meta.hpp
[the remark this answers is not in the document]
Comment (jbbjarnason) [reply libs/dbus_util/inc/public/tfc/dbus/sdbusplus_meta.hpp:1]: added in `dbus_util_test.cpp` but they are bloated as hell

File: libs/ipc/inc/public/tfc/ipc/details/impl.hpp [hunk 7]
```diff
@@ -112,22 +112,17 @@ class signal : public transmission_base<type_desc>, public std::enable_shared_fr
...
-            }
-            case state_e::complete: {
-              self.complete(err, bytes_sent);
-              break;
-            }
+          if (first_call) {
+            first_call = false;
+            azmq::async_send(socket, asio::buffer(*buffer), std::move(self));
+          } else {
+            self.complete(err, bytes_sent);
```
[the remark this answers is not in the document]
Comment (jbbjarnason) [reply libs/ipc/inc/public/tfc/ipc/details/impl.hpp]: reverted

File: libs/ipc/src/dbus_client_iface.cpp (hunk 3 above)
Comment (jbbjarnason) [reply libs/ipc/src/dbus_client_iface.cpp:74]: ditto

File: exes/mqtt-bridge/src/external_to_tfc.cpp [hunk 8]
```diff
@@ -44,6 +44,8 @@ auto external_to_tfc<ipc_client_t, config_t, signal_v>::receive_new_value(
           // to the signal
           if constexpr (std::is_same_v<value_t, int64_t>) {
             signal.send(static_cast<int64_t>(std::get<uint64_t>(value)));
+          } else if constexpr (tfc::stx::is_expected_quantity<value_t>) {
+            // todo
```
[the remark this answers is not in the document]
Comment (omarhogni) [reply exes/mqtt-bridge/src/external_to_tfc.cpp:48]: issue created.

File: libs/ipc/inc/public/tfc/ipc/glaze_meta.hpp [hunk 9]
```diff
@@ -37,9 +37,31 @@ struct meta<tfc::ipc::details::type_e> {
...
+struct meta<tfc::ipc::details::mass_error_e> {
+  using enum tfc::ipc::details::mass_error_e;
+  // clang-format off
+  static auto constexpr value{ glz::enumerate(
+    "no_error", no_error, "Not an error",
+    "cell_fault", cell_fault, "Load cell fault",
+    "module_fault", module_fault, "General weigher module fault",
+    "power_failure", power_failure, "Supply voltage too low",
+    "over_range", over_range, "Over max value",
+    "under_range", under_range, "Under minimum value",
```
Comment (omarhogni) [inline libs/ipc/inc/public/tfc/ipc/glaze_meta.hpp:56]:
```suggestion
    "over_range", over_range, "Over range",
    "under_range", under_range, "Under range",
```

**Review by omarhogni — APPROVED** [review]

**Pull request merged**